cpp/src/arrow/compute/kernels/vector_selection_take_internal.cc

```diff
@@ -395,10 +526,45 @@ struct FixedWidthTakeImpl {
     out_arr->null_count = out_arr->length - valid_count;
     return Status::OK();
   }
+
+  static Status ChunkedExec(KernelContext* ctx, const ChunkedArray& values,
+                            const ArraySpan& indices, ArrayData* out_arr,
+                            int64_t factor) {
+    const bool out_has_validity = values.null_count() > 0 || indices.MayHaveNulls();
+
+    ChunkedFixedWidthValuesSpan chunked_values{values};
+    ResolvedIndicesState resolved_idx;
+    RETURN_NOT_OK(resolved_idx.InitWithIndices<IndexCType>(
+        /*chunks=*/values.chunks(), /*idx_length=*/indices.length,
+        /*idx=*/indices.GetValues<IndexCType>(1), ctx->memory_pool()));
+
+    int64_t valid_count = 0;
+    arrow::internal::GatherFromChunks<kValueWidthInBits, IndexCType, WithFactor::value>
```

Comment:
As a further optimization, perhaps we should test whether `chunk_index_vec` contains a single distinct chunk index, in which case we could similarly fallback on ArraySpan-take (I'm not sure how expensive the check would be, though).

Probably for another PR/issue, if any.

Reply:
This is the kind of optimization that depends on statistical properties of the input data so gains are hard to quantify. Here I went with the assumption that if there are multiple chunks, values are gathered from multiple of them.